cmd/fyne/internal/commands/package.go

```diff
@@ -302,7 +306,14 @@ func (p *Packager) removeBuild(files []string) {
 	}
 }
 
-func (p *Packager) validate() error {
+func (p *Packager) validate() (err error) {
+	p.tempDir, err = ioutil.TempDir("", "fyne-package-*")
+	defer func() {
+		if err != nil {
+			_ = os.RemoveAll(p.tempDir)
+		}
+	}()
```

Comment:
I am not sure that this does what is intended. It will not remove the temporary directory if it gets an error anywhere (not just if the `ioutil.TempDir("", "fyne-package-*")` call failed). It might be better to do this instead:
```suggestion
	p.tempDir, err = ioutil.TempDir("", "fyne-package-*")
	if err != nil {
		return err
	}
	defer os.RemoveAll(p.tempDir)
```

Reply:
Sadly I cannot do that as it should only delete if there was an error.
My understanding is that because `err` is a return var it will be evaluated during the defer as the final value of `err` and so therefore if a later err is set it will catch that too...

Reply:
Ah, I see. That is how it works as I have understood it as well. I just thought you meant to do something else.
If it only removes the tempdir on error, does this mean that the tempdir fills up every time you do a successful package?

Reply:
The directory is also deleted after a successful package.
I would have preferred a closer "create then delete" pairing of code, but we need it in validate and also package, which are called directly by other modes as well (release, install)